cs/src/Management/TunnelManagementClient.cs

```diff
@@ -140,14 +163,23 @@ public class TunnelManagementClient : ITunnelManagementClient
         /// <see cref="HttpClientHandler"/> specified (or at the end of the chain) must have
         /// automatic redirection disabled. The provided HTTP handler will not be disposed
         /// by <see cref="Dispose"/>.</param>
+        /// <param name="apiVersion"> Api version to use for tunnels requests, accepted
+        /// values are <see cref="TunnelsApiVersions"/></param>
         public TunnelManagementClient(
             ProductInfoHeaderValue[] userAgents,
             Func<Task<AuthenticationHeaderValue?>>? userTokenCallback = null,
             Uri? tunnelServiceUri = null,
-            HttpMessageHandler? httpHandler = null)
+            HttpMessageHandler? httpHandler = null,
+            string? apiVersion = null)
         {
             Requires.NotNullEmptyOrNullElements(userAgents, nameof(userAgents));
             UserAgents = Requires.NotNull(userAgents, nameof(userAgents));
+            if (!TunnelsApiVersions.Contains(apiVersion) && !string.IsNullOrEmpty(apiVersion))
```

Comment:
nit: first check for `IsNullOrEmpty` and then `Contains` so that `Contains` doesn't need to run for null `apiVersion`

Reply:
updated